numpy/core/src/private/ufunc_override.c

```diff
@@ -8,148 +8,172 @@
 
 #include "ufunc_override.h"
 
-static void
+
+static int
 normalize___call___args(PyUFuncObject *ufunc, PyObject *args,
-                    PyObject **normal_args, PyObject **normal_kwds,
-                    int nin)
+                        PyObject **normal_args, PyObject **normal_kwds,
+                        int nin, int nout)
```

Comment:
I'm somewhat concerned that upcasting a pointer is allowed to invoke undefined behaviour in a C compiler, so I think it's very important we change that to `PyObject*` until we work out a better way to indicate `matmul` and `dot`